Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/audit/COREX12BatchAuditTransforms.java

```diff
@@ -0,0 +1,297 @@
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND 
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT 
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS 
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+ */
+package gov.hhs.fha.nhinc.corex12.docsubmission.audit;
+
+import com.services.nhinc.schema.auditmessage.AuditMessageType;
+import com.services.nhinc.schema.auditmessage.AuditSourceIdentificationType;
+import com.services.nhinc.schema.auditmessage.CodedValueType;
+import com.services.nhinc.schema.auditmessage.EventIdentificationType;
+import com.services.nhinc.schema.auditmessage.ParticipantObjectIdentificationType;
+import gov.hhs.fha.nhinc.common.auditlog.LogEventRequestType;
+import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
+import gov.hhs.fha.nhinc.common.nhinccommon.NhinTargetSystemType;
+import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
+import gov.hhs.fha.nhinc.nhinclib.NullChecker;
+import gov.hhs.fha.nhinc.transform.audit.AuditDataTransformConstants;
+import gov.hhs.fha.nhinc.transform.audit.AuditDataTransformHelper;
+import gov.hhs.fha.nhinc.transform.audit.XDRTransforms;
+import org.apache.log4j.Logger;
+import org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeBatchSubmission;
+import org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeBatchSubmissionResponse;
+
+/**
+ * Class provides services to Construct Audit Messages from the X12 Batch request and response
+ *
+ * @author svalluripalli
+ */
+public class COREX12BatchAuditTransforms {
+
+    private static final Logger LOG = Logger.getLogger(COREX12BatchAuditTransforms.class);
+    private final XDRTransforms oXDR = new XDRTransforms();
+    private final COREX12RealtimeTransforms oCoreRealTimeTransforms = new COREX12RealtimeTransforms();
+
+    /**
+     * Transforms the Batch Request message to Audit logging specific message.
+     *
+     * @param request
+     * @param assertion
+     * @param target
+     * @param direction
+     * @param _interface
+     * @return LogEventRequestType
+     */
+    public LogEventRequestType transformBatchRequestToAuditMsg(COREEnvelopeBatchSubmission request, AssertionType assertion, NhinTargetSystemType target, String direction, String _interface, boolean isRequesting) {
+        LogEventRequestType result = null;
+        AuditMessageType auditMsg = null;
+        LOG.trace("Begin COREX12RealtimeTransforms -> transformRequestToAuditMsg() -- NHIN");
+        if (request == null) {
+            LOG.error("Requst Object was null");
+            return null;
+        }
+
+        if (assertion == null) {
+            LOG.error("Assertion was null");
+            return null;
+        }
+
+        // check to see that the required fields are not null
+        boolean missingReqFields = areRequiredCOREX12BatchFiledsNull(request, assertion);
```

Comment:
No need to declare a separate variable; use rhs inside the if condition directly.